pageserver/src/tenant.rs

```diff
@@ -422,13 +422,46 @@ impl Tenant {
             init_order,
             CreateTimelineCause::Load,
         )?;
-        let new_disk_consistent_lsn = timeline.get_disk_consistent_lsn();
+        let disk_consistent_lsn = timeline.get_disk_consistent_lsn();
+        assert_eq!(
+            disk_consistent_lsn,
+            up_to_date_metadata.disk_consistent_lsn(),
+            "these are used interchangeably; need to move layer map loading before Timeline creation to avoid drifting"
+        );
```

Comment:
I've kept this, because at least I found it confusing, but earlier there were the separate two step process which also shared this value. This is expectation which has always been there, but you are assuming the sprawling nest of Timeline struct creating functions called from above copy this correctly, and we are able to read it back as intended :)